src/XMakeBuildEngine/Evaluation/LazyItemEvaluator.IncludeOperation.cs

```diff
@@ -58,6 +58,19 @@ protected override ICollection<I> SelectItems(ImmutableList<ItemData>.Builder li
                     }
                 }
 
+                var excludePatternsForGlobs = new Lazy<ISet<string>>(() =>
```

Comment:
Yeah it avoids extra object creation for the lazy object itself and possibly the closure used for the creation delegate.